sympy/matrices/random.py

```diff
@@ -0,0 +1,2421 @@
+from ..core import I as _i
+from ..core.mul import Mul as _multiply
+from ..core.random import sample
+from ..core.sympify import _sympify
+from ..functions import sqrt as _sqrt, re as _re, im as _im, \
+    transpose as _t, adjoint as _c
+from ..tensor import shape as _shape
+from .dense import eye as _eye
+from .inverse import _inv
+
+__all__ = 'projection', 'jordan', 'transposition', \
+          'permutation', 'elementary', 'rotation', 'reflection', \
+          'diagonal_normal', 'jordan_normal', 'isometry_normal', \
+          'triangular', 'invertible', 'singular', \
+          'idempotent', 'nilpotent', 'diagonalizable', 'trigonalizable', \
+          'orthogonal', 'unitary', 'normal', \
+          'symmetric', 'hermite', 'square', \
+          'regular_to_singular', 'complex_to_real'
+
+
+_ALT = False
```

Comment:
There should be some comment explaining what `_ALT` is. I've looked through the code and I don't understand its purpose.

Probably this variable should just be removed. If there is a reason to choose between the different codepaths then it should not be done using a global variable.

Reply:
Good point. Removed `_ALT` and moved alternative implementations to `test_random.py` as it is used in testing different codepaths.